sources/metricdog/src/metricdog.rs

```diff
@@ -0,0 +1,158 @@
+use crate::config::Config;
+use crate::error::{self, Result};
+use crate::service_check::ServiceCheck;
+use bottlerocket_release::BottlerocketRelease;
+use log::debug;
+use reqwest::blocking::Client;
+use snafu::ResultExt;
+use std::collections::HashMap;
+use std::str::FromStr;
+use std::time::Duration;
+use url::Url;
+
+/// The send function optionally takes a timout parameter so that we can have a short timeout for
+/// `boot_success`. When `None` is passed, the default timeout is used. 20 seconds was arbitrarily
+/// chosen and can be changed if the need arises.
+const DEFAULT_TIMEOUT_SECONDS: u64 = 20;
+
+/// Sends key-value pairs as query params to a URL configured in `config`. Also provides the ability
+/// to check the health of a list of services and send information about whether or not the services
+/// are running.
+pub(crate) struct Metricdog {
+    /// The `Metricdog` configuration, e.g. from `/etc/metricdog.toml`
+    config: Config,
+    /// Information about the Bottlerocket release, e.g. from `os-release`
+    os_release: BottlerocketRelease,
+    /// A trait object that checks if a service (listed in `config`) is healthy. This can be passed-
+    /// in, but defaults to an object that uses `systemctl` to check services.
+    healthcheck: Box<dyn ServiceCheck>,
+    /// The metrics_url, having been parsed during construction of the `Metricdog` object.
+    metrics_url: Url,
+}
+
+impl Metricdog {
+    /// Create a new instance by optionally passing in the `Config`, `BottlerocketRelease`, and
+    /// `ServiceCheck` objects. For each of these, if `None` is passed, then the default is used.
+    pub(crate) fn from_parts(
+        config: Config,
+        os_release: BottlerocketRelease,
+        healthcheck: Box<dyn ServiceCheck>,
```

Comment:
I'm not sure I understand the comment, since none of these fields are `Option`..?

Reply:
An example of "comments are lies." Fixed in [b8b5c3a](https://github.com/bottlerocket-os/bottlerocket/compare/153acff977082318fef8fc8e61024b26574c362e..b8b5c3ad92298db2c46262eb2b6c5b6ea3f60187)